src/metabase/models/user.clj

```diff
@@ -91,8 +93,14 @@
      {:locale (i18n/normalized-locale-string locale)})))
 
 (t2/define-after-insert :model/User
-  [{user-id :id, superuser? :is_superuser, :as user}]
+  [{user-id :id, superuser? :is_superuser, settings :settings, :as user}]
   (u/prog1 user
+    (let [current-version (:tag config/mb-version-info)]
+      (log/info (trs "Setting User {0}''s last_acknowledged_version to {1}, the current version" user-id current-version))
+      ;; Can't use mw.session/with-current-user-id due to circular require
+      (binding [api/*current-user-id*       user-id
+                setting/*user-local-values* (delay (atom (or settings {})))]
+        (setting/set! :last-acknowledged-version current-version)))
```

Comment:
This is so frustrating

Reply:
Yeah this is awkward af to read and write. I'm not a fan of User or Database settings but it is what it is for now